### ISSUE TITLE
basemap pcolormesh warning with matplotlib 2.0

### ISSUE THREAD

**Issue by adamabernathy** [matplotlib/matplotlib#9249]
### Bug report

**Bug summary**

When using `pcolormesh` in `basemap` (anaconda install) the following error is often thrown:

```shell
/Users/blah/anaconda2/lib/python2.7/site-packages/mpl_toolkits/basemap/__init__.py:3413: MatplotlibDeprecationWarning: The ishold function was deprecated in version 2.0.
  b = ax.ishold()
/Users/blah/anaconda2/lib/python2.7/site-packages/mpl_toolkits/basemap/__init__.py:3422: MatplotlibDeprecationWarning: axes.hold is deprecated.
    See the API Changes document (http://matplotlib.org/api/api_changes.html)
    for more details.
  ax.hold(b)
```

I suppose this is mostly due to basemap not being updated to matplotlib 2.0's api.  The fix that i've got now seems to be working in all the cases in which I use `pcolormesh` is to comment out (or just remove) any reference to `ax.hold`.  Clearly since this is throwing an error and still working I'd say its safe to just remove it.

https://matplotlib.org/devdocs/api/_as_gen/matplotlib.axes.Axes.hold.html

```python
def pcolormesh(self,x,y,data,**kwargs):
    """
    Make a pseudo-color plot over the map
    (see matplotlib.pyplot.pcolormesh documentation).

    If ``latlon`` keyword is set to True, x,y are intrepreted as
    longitude and latitude in degrees.  Data and longitudes are
    automatically shifted to match map projection region for cylindrical
    and pseudocylindrical projections, and x,y are transformed to map
    projection coordinates. If ``latlon`` is False (default), x and y
    are assumed to be map projection coordinates.

    Extra keyword ``ax`` can be used to override the default axis instance.

    Other \**kwargs passed on to matplotlib.pyplot.pcolormesh.
    """
    ax, plt = self._ax_plt_from_kw(kwargs)
    # allow callers to override the hold state by passing hold=True|False
    # b = ax.ishold()
    # h = kwargs.pop('hold',None)
    # if h is not None:
    #     ax.hold(h)
    try:
        ret =  ax.pcolormesh(x,y,data,**kwargs)
    except:
        # ax.hold(b)
        raise
    # ax.hold(b)
    # reset current active image (only if pyplot is imported).
    if plt:
        plt.sci(ret)
    # clip for round polar plots.
    if self.round: ret,c = self._clipcircle(ax,ret)
    # set axes limits to fit map region.
    self.set_axes_limits(ax=ax)
    if self.round:
        # for some reason, frame gets turned on.
        ax.set_frame_on(False)
    return ret
```

**Comment by tacaswell** [comment]
To verify, the code works but emits warnings?

**Comment by adamabernathy** [comment]
Correct.

**Comment by tacaswell** [comment]
attn @WeatherGod

**Comment by WeatherGod** [comment]
This should have been fixed for v1.1.0, which is available on conda-forge.

On Thu, Sep 28, 2017 at 12:27 PM, Thomas A Caswell <notifications@github.com

**Comment by adamabernathy** [comment]
I installed it this morning from conda-forge.

**Comment by WeatherGod** [comment]
what does `python -c "from mpl_toolkits import basemap;
print(basemap.__version__)"` outputs?

**Comment by adamabernathy** [comment]
```shell
$: python -c "from mpl_toolkits import basemap;
> print(basemap.__version__)"
1.0.7
```

**Comment by WeatherGod** [comment]
Yeah, so it is still using the older version. This sometimes happens if you
have a mix of the pip-installed version and a conda-installed version.
namespace packages are a PITA. Sometimes, the only way to deal with them
properly, especially in older installs of namespace packages is to just
manually delete them out of your site-packages directory.

**Comment by WeatherGod** [comment]
I should note... watch your output for `conda update -c conda-forge
basemap` closely. I just recently ran into a situation where I had a
somewhat older install of conda, and it ignored my "-c conda-forge" option,
saying that it couldn't find that channel, and then happily went ahead and
suggested an update using the standard channel.

Unfortunately, the upgrade path for conda from that old version to current
seems to be extremely painful (just had to do it manually on two test
machines). I wish I had just wiped conda and installed fresh, instead.
